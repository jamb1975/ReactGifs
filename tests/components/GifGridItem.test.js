import { render, screen, fireEvent } from "@testing-library/react";
import { GifGridItem } from '../../src/components/GifGridItem'

describe('Tests on <GifExpertApp />', () => {
  
  const title = "Genesis";
  const url = "https://media4.giphy.com/media/v1.Y2lkPTUwZTdiYzA0OG40amYyM3cyYXBha3hxY3BkdHo5c29iNzV5bWhwd203Zm5jbzh3diZlcD12MV9naWZzX3NlYXJjaCZjdD1n/rhdscFKah6Rva/giphy.gif";  
 
  test('Verificar snapshot', () =>{
      const { container } = render(<GifGridItem  title = {title} url ={ url } />);
      expect( container ).toMatchSnapshot();
        
  });

  
  test('debe mostrar el url y title', () =>{
       render(<GifGridItem  title = {title} url ={ url } />);
       const { src, alt } = screen.getByRole('img');
       expect(src).toBe( url );
       expect(alt).toBe( title );
        
  });

  test('debe de tener el titulo', () =>{
       render(<GifGridItem  title = {title} url ={ url } />);
       expect(screen.getByText(title)).toBeTruthy();
  });

})