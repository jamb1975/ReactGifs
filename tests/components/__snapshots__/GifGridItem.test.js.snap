// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Tests on <GifExpertApp /> Verificar snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Genesis"
      src="https://media4.giphy.com/media/v1.Y2lkPTUwZTdiYzA0OG40amYyM3cyYXBha3hxY3BkdHo5c29iNzV5bWhwd203Zm5jbzh3diZlcD12MV9naWZzX3NlYXJjaCZjdD1n/rhdscFKah6Rva/giphy.gif"
    />
    <p>
      Genesis
    </p>
  </div>
</div>
`;
